### PR TITLE
Update transaction wire format, toward credit-only accounts

### PR DESCRIPTION
**Pull request by CriesofCarrots**
Also:
- Add preliminary handling for designating credit-only accounts (currently, solana handles credit-only accounts the same as credit-debit accounts; more tweaking may be needed here when this changes)
- Repair work to get CI passing again

### CODE REVIEW

**Comment by CriesofCarrots** [comment]
These test failures only happen when run against localnet (using docker edge image of solana), not when run against a local solana cluster started manually with `.run.sh`.
Will examine more after next solana docker publish...

File: src/transaction.js [hunk 1]
```diff
@@ -109,7 +113,7 @@ export class Transaction {
   /**
    * A recent transaction id.  Must be populated by the caller
    */
-  recentBlockhash: ?Blockhash;
+  recentBlockhash: Blockhash;
```
Comment (mvines) [inline src/transaction.js]: `Blockhash|null` perhaps?  There won't always be a value here I think

**Comment by mvines** [comment]
Looks great overall.  Man WASM/Rust would be nice one day :)

**Comment by CriesofCarrots** [comment]
@mvines , okay to merge?
Sorry it ended up to be such a motley collection of commits...
Incidentally, it looks like example-tictactoe and example-messagefeed will both need to run on `dist: xenial` in order to successfully build bpf in CI. Not sure if that will have any negative ramifications.

**Comment by mvines** [comment]
Awesome, thanks!
I don't see any issue with switching the two examples over to xenial

**Pull request merged**